src/compile/mark/text.ts

```diff
@@ -79,16 +79,25 @@ export namespace text {
 
     // text
     if (model.has(TEXT)) {
+      let textTemplate = model.field(TEXT);
+      let datum = model.field(TEXT);
       if (contains([QUANTITATIVE, TEMPORAL], model.fieldDef(TEXT).type)) {
         const format = model.config().mark.format;
-        extend(p, formatMixins(model, TEXT, format));
-      } else {
-        p.text = { field: model.field(TEXT) };
+        datum = formatMixins(model, TEXT, format).text.template;
+      }
+      if (fieldDef.unit) {
+        if (!fieldDef.unitPosition || fieldDef.unitPosition == 'suffix') {
```

Comment:
The defaults should be on the config. Check out how default configs are being used and mirror it for unitPosition.